src/components/button/button.d.ts

```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+export interface ButtonProps {
```

Comment:
Did you consider use types instead of interfaces?

Reply:
I have, my understanding is that either works fine as long as we're consistent throughout Carbon but that you can't merge declarations when using `type` and I can see scenarios where a developer may want to extend the definitions.  I'll talk to Andrew about it when he gets in as I chose to go with `interface` to allow for this, but will change it to `type` if it's not needed.

Reply:
I've spoke to him, we're going to leave it as interface for now so that anyone using Carbon can extend the definition if they need to

Reply:
I understand.
I've found it https://stackoverflow.com/questions/41385059/possible-to-extend-types-in-typescript but it's OK for me. As you said `as long as we're consistent`. :)